synapse/storage/database.py

```diff
@@ -267,8 +267,7 @@ def execute_batch(self, sql: str, args: Iterable[Iterable[Any]]) -> None:
 
             self._do_execute(lambda *x: execute_batch(self.txn, *x), sql, args)
         else:
-            for val in args:
-                self.execute(sql, val)
+            self.executemany(sql, args)
```

Comment:
Is it expected that this has any performance difference?

Reply:
The docs are unclear, so possibly? Either way probably clearer to shove it down a layer :shrug:

Reply:
Hopefully the underlying code is optimized, yeah. 👍